libraries/AP_Compass/AP_Compass_Calibration.cpp

```diff
@@ -316,16 +313,18 @@ uint8_t Compass::handle_mag_cal_command(const mavlink_command_long_t &packet)
         uint8_t mag_mask = packet.param1;
         
         if (mag_mask == 0) { // 0 means all
-            if(!accept_calibration_all()) {
+            if(!_accept_calibration_mask(0xFF)) {
                 result = MAV_RESULT_FAILED;
             }
-            break;
         }
         
-        if(!accept_calibration_mask(mag_mask)) {
+        if(!_accept_calibration_mask(mag_mask)) {
             result = MAV_RESULT_FAILED;
         }
-        break;
+
+        for (int i = 0; i < COMPASS_MAX_INSTANCES; i++) {
+            _calibrator[i].clear();
```

Comment:
That was the idea, I was interprating the ACCEPT to say anything that isn't accepted should be discarded, but I can see your point that might not be the way others would read it. I can modify it, but it means passing down a flag to accept_calibration() to mark if a calibrator should be cleared or not, which I wasn't a big fan of. But thats probably still better.